fastlane/Fastfile

```diff
@@ -1420,4 +1425,43 @@ platform :ios do
     # Return the formatted build code
     BUILD_CODE_FORMATTER.build_code(version: build_code_next)
   end
+
+  def remove_app_clip_dependency!
+    project_path = '../podcasts.xcodeproj'
+    project = Xcodeproj::Project.open(project_path)
+
+    app_target_name = 'podcasts'
+    app_target = project.targets.find { |t| t.name == app_target_name }
+    UI.user_error!("Could not find target named #{app_target_name} in #{project_path}") if app_target.nil?
+
+    app_clip_target_name = 'Pocket Casts App Clip'
+    app_clip_target = project.targets.find { |t| t.name == app_clip_target_name }
+    UI.user_error!("Could not find clip_target named #{app_clip_target_name} in #{project_path}") if app_clip_target.nil?
+
+    app_clip_dependency_reference = app_target.dependencies.find { |d| d.target == app_clip_target }
+    UI.user_error!("Target #{app_target.name} does not have #{app_clip_target.name} as a dependency.") if app_clip_dependency_reference.nil?
+
+    UI.message("Removing #{app_clip_dependency_reference.target.name} as a dependency of #{app_target.name}...")
+    UI.verbose("Dependencies count before deletion #{app_target.dependencies.count}.")
+    app_target.dependencies.delete(app_clip_dependency_reference)
+    UI.verbose("Dependencies count after deletion #{app_target.dependencies.count}.")
+
+    embed_app_clips_phase_name = 'Embed App Clips'
+    embed_app_clips_phase = app_target.build_phases.find { |b| b.display_name == 'Embed App Clips' }
+    UI.user_error!("#{app_target.name} has no build phase with display name #{embed_app_clips_phase_name}") if embed_app_clips_phase.nil?
+
+    app_clip_path = 'Pocket Casts App Clip.app'
+    app_clip_file_reference = embed_app_clips_phase.files_references.find { |r| r.path == app_clip_path }
+    UI.user_error!("The #{embed_app_clips_phase.display_name} build phase in #{app_target.name} has no file reference with path #{app_clip_path}") if app_clip_file_reference.nil?
+
+    UI.message("Removing #{app_clip_file_reference.path} from #{embed_app_clips_phase.display_name}...")
+    UI.verbose("Paths to embed count before deletion #{embed_app_clips_phase.files_references.count}.")
+    embed_app_clips_phase.remove_file_reference(app_clip_file_reference)
+    UI.verbose("Paths to embed count after deletion #{embed_app_clips_phase.files_references.count}.")
+
+    UI.message('Saving the project file....')
+    project.save
+
+    UI.success('Dependency successfuly removed.')
+  end
```

Comment:
See the improvement suggestions I made for this script in https://github.com/Automattic/pocket-casts-ios/commit/a89eeea1dbd420e6c04601ea3075fd34da647183#commitcomment-151007623 before you opened this PR 😅 

This is not really a blocker (as long as your code works…). But I think my suggestions are still worth it given that it would (1) simplify the code a bit (just using `delete_if` instead of having the same be done with multiple lines of Ruby) and (2) be more flexible and generic (avoid hardcoding the target's names)

Reply:
Thanks @AliSoftware ! That's definitely simpler and easier to read.

I came to the implementation in this PR iteratively, sort of using all those logs to figure out what step to take next. I do agree it's a lot the way it is now... At the same time, I like how it would fail if the project is not set up with an App Clip, to signify that something has changed in the setup the automation expects and the user should act accordingly.

I'm going to merge as is just to unblock future builds, then follow up with a PR that will try to take the best of both implementations. Thanks again!